CONTRIBUTING.md

```diff
@@ -0,0 +1,129 @@
+violations, shoot out an email to founders@gitcoin.co
+
+The Code of Conduct is designed and intended, above all else, to help establish
+a culture within the project that allows anyone and everyone who wants to
+contribute to feel safe doing so.
+
+Open, diverse, and inclusive communities live and die on the basis of trust.
+Contributors can disagree with one another so long as they are done in good
+faith and everyone is working towards a common goal.
+
+## Issues
+Issues in `gitcoin/web` are the primary means by which bug reports and
+general discussions are made. An contributor is allowed to create an issue,
+discuss and provide a fix if needed.
+
+## Discussions And General Help
+As Gitcoin is still at it's early stages, drop by [gitcoin.co/slack](gitcoin.co/slack)
+and say hi to know what's next / to get your answers
+cleared up.
+
+## Pull Requests
+Pull Requests are the way in which concrete changes are made to the code and
+documentation.
+
+### Step 1: Fork
+
+Fork the project [on GitHub](https://github.com/gitcoinco/web) and clone your
+fork locally.
+
+```text
+$ git clone git@github.com:username/web.git
+$ cd web
+$ git remote add upstream https://github.com/gitcoinco/web.git
+$ git fetch upstream
+```
+
+### Step 2: Branch
+
+It's always better to create local branches to work on a specific issue. Makes
+life easier for you if you are the kind who enjoys multiple things parallely.
+These should also be created directly off of the `master` branch.
+
+```text
+$ git checkout -b my-branch -t upstream/master
+```
+
+### Step 3: Code
+
+As of now, we don't have any sort of design style / lint to validate things.
+So we ask you to ensure all these are met before you shoot out a PR.
+- Avoid trailing whitespace & un-necessary white lines
+- Indentation is as follows
+  - 1 tab = 2 spaces for `.html` files
+  - 1 tab = 4 spaces for everything else
+  - __Linters will be added to make this easier.__
+
+### Step 4: Commit
+
+1. List all your changes as a list if needed else simply give a brief
+  description on what the changes
+2. All lines at 100 columns.
+3. If your PR fixed an issue, Use the `Fixes:` prefix and the full issue URL.
+  For other references use `Refs:`.
+
+   __Examples:__
+   - `Fixes: https://github.com/gitcoinco/web/issues/87`
+   - `Refs: https://github.com/gitcoinco/web/issues/91`
+
+4. __Sample commit A__
+   ```txt
+   if you can write down the changes explaining it in a paragraph which each
+   line wrapped within 100 lines.
+
+   Fixes: https://github.com/gitcoinco/web/issues/87
+   Refs: https://github.com/gitcoinco/web/issues/91
+   ```
+
+   __Sample commit B__
+   ```txt
+   - list out your changes as points if there are many changes
+   - if needed you can also send it across as
+   - all wrapped within 100 lines
+
+   Fixes: https://github.com/gitcoinco/web/issues/87
+   Refs: https://github.com/gitcoinco/web/issues/91
+   ```
+5. Ensure your squash your commit to make our history neater and reviewing
```

Comment:
do we want to do this? ive worked on teams that have gone etiher way and i dont feel like we've derived a lot of value from this approach.  my 2c

Reply:
Well we aren't forcing them to squash it, but making it neater is always welcome.
I have folks who submit 6 PRs which could be actually just 2 commits!  
It's just about keeping the history neater I guess.

How about rephrasing it to  ?
```
Squashing commit to make the history neater are always welcomed
``` 
@Elaniobro your thoughts?

PS: yeah i noticed the spelling & grammar error. will fix that

Reply:
Squash and Merge FTW! Yea, all for it.

Good read: https://chris.beams.io/posts/git-commit/#seven-rules